src/writer/proposal.ts

```diff
@@ -223,5 +223,16 @@ export async function action(body, ipfs, receipt, id): Promise<void> {
   const query = 'INSERT IGNORE INTO proposals SET ?; ';
   const params: any[] = [proposal];
 
-  await db.queryAsync(query, params);
+  const result = await db.queryAsync(query, params);
+
+  if (result.affectedRows > 0) {
+    await db.queryAsync(
+      `
+      INSERT INTO user_space (space_id, user_id, proposals_count)
+      VALUES(?, ?, 1)
+      ON DUPLICATE KEY UPDATE proposals_count = proposals_count + 1
+      `,
+      [space, author]
+    );
+  }
```

Comment:
Could we merge all queries into one query, I worry about DB network errors after executing the first one and failing on the second one. then we will have wrong counts

Reply:
How would you merge it?

the second one depends on the result of the first one, as we should confirm there’s really a new row, else we skip the counter update.

the lack of transaction and counters un syncing has been raised, and @less has acknowledged that it’s an inconvenient we can leave with

Reply:
Failure, or duplicate

Reply:
But even if we merge, if the first query fails, the second one also fails no?

Reply:
Yes, unless it throws an exception. Second query executes only if there is one successful insert